gorabbitmq_test.go

```diff
@@ -0,0 +1,1669 @@
+package gorabbitmq_test
```

Comment:
the tests in this package really seem like a lot, I think you got a bit lost in those table driven tests

it's really hard to read those, let alone understand what is happening
maybe check if you can streamline this a bit more

maybe this is also not the best case for table driven tests and you would do better with a simple test.Main function that does the setup and teardown